### PR TITLE
Bear trap fixes

### PR DESCRIPTION
**Pull request by boy2mantwicethefam**
More bear trap fixes
Thanks @ShiftyRail for testing and additional fixes

:cl:
 * tweak: You can no longer add lit IEDs into bear traps
 * bugfix: Fixed a bunch of bear trap bugs

### CODE REVIEW

**Comment by boy2mantwicethefam** [comment]
Don't merge this yet, I'm fixing other stuff

**Comment by boy2mantwicethefam** [comment]
Fixed a bug where animals wouldn't trigger the trap and get damaged
IED overlays now get removed when the IED detonates, not at any point the bear trap triggers
Walking over an IED'd bear trap won't trigger the bear trap anymore
Moved update_icon to the end of the cross check because it only checks if the bear trap is still armed
Updated localization files

**Comment by boy2mantwicethefam** [comment]
"horrifying fiery dragon" self-drugged message when triggering a bear trap now appears when triggering an IED'd bear trap instead of the other way around where a crocodile would have showed up when triggering an IED'd bear trap

**Comment by N3X15** [comment]
image
I should make a beer coaster out of that

**Comment by ShiftyRail** [comment]
Thank you Barry you are my greatest ally

**Comment by boy2mantwicethefam** [comment]
If there is no leg for the bear trap to hatch on anymore, breaking out of bear traps should be instantaneous now
Removed default definition for the variable because it is set already when the trap is triggered and such (there are some oversights in the code but I can't be arsed to fix them yet)
You can't shove a lit IED into a bear trap anymore
The IED'd bear trap checks if it has a leg to hatch on

As a note, things like trapping people on vehicles is already handled by on_foot which checks if the victim is already locked onto something
Scrutinize the code, my work is done, wake me up when you find out something terribad, I didn't test any of this and I won't unless it's absolutely necessary to even get the PR merged in the first place

**Comment by ShiftyRail** [comment]
Don't worry I'll test this for you to atone for my coder sins

File: code/defines/obj/weapon.dm [hunk 1]
```diff
@@ -412,7 +412,7 @@
 	var/breakouttime = 60
 	var/armed = 0
 	var/trapped = 0
-	var/datum/organ/external/trappedorgan = "LIMB_LEFT_LEG"
+	var/datum/organ/external/trappedorgan //The limb currently trapped, it must be a leg
```
Comment (Kurfursten) [inline code/defines/obj/weapon.dm:415]: iirc it can be an arm if you try to touch with hand

File: code/defines/obj/weapon.dm [hunk 2]
```diff
@@ -479,62 +479,62 @@
...
 					var/area/A = get_area(bombturf)
 					var/log_str = "[key_name(usr)]<A HREF='?_src_=holder;adminmoreinfo=\ref[user]'>?</A> has rigged a beartrap with an IED at <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[bombturf.x];Y=[bombturf.y];Z=[bombturf.z]'>[A.name] (JMP)</a>."
 					message_admins(log_str)
 					log_game(log_str)
+					IED = I
 					to_chat(user, "<span class='notice'>You sneak the [IED] underneath the pressure plate and connect the trigger wire.</span>")
 					desc = "A trap used to catch bears and other legged creatures. <span class='warning'>There is an IED hooked up to it.</span>"
 					overlays.Add(ied_overlay)
+			if(3) //There is a whole thing about setting the IED's detonation status for IED'd bear traps, this is to avoid insanity
+				to_chat(user, "<span class='warning'It's dangerous to shove a lit IED into a bear trap! Find another way to get rid of it!</span>")
```
Comment (Kurfursten) [inline code/defines/obj/weapon.dm:535]: cl this

Comment (boy2mantwicethefam) [reply code/defines/obj/weapon.dm:535]: Done

File: code/defines/obj/weapon.dm (hunk 1 above)
Comment (boy2mantwicethefam) [reply code/defines/obj/weapon.dm:415]: Checked attack_hand, it can't, it only damages your arm and then deactivates itself

**Comment by ShiftyRail** [comment]
Did the tests, `trap` does NOT get called on Crossed.
I'll try to make a PR to that branch to fix it. I apologise to the c*llab god for my sins and I'll do 5 bugfixes for atonement.

**Comment by ShiftyRail** [comment]
IEDs also don't properly attach, they cause a runtime. Will fix too

**Comment by ShiftyRail** [comment]
https://github.com/boy2mantwicethefam/vgstation13/pull/1

@boy2mantwicethefam

**Comment by Kurfursten** [comment]
Is this still DNM?

**Comment by boy2mantwicethefam** [comment]
no, I mentioned my work is done

**Pull request merged**